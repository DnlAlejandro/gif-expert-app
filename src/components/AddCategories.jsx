import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategories = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState("");

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        //setCategories((categories) => [inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setInputValue("");
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="search gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    );
};

AddCategories.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
    
}