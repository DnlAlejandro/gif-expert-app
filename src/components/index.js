export * from './AddCategories';
export * from './GifGrid';
export * from './GifItem';