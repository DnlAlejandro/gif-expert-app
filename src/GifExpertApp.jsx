import { useState } from "react";
import { AddCategories, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(["UFC"]);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);
    };

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategories
                //setCategories={setCategories}
                onNewCategory={onAddCategory}
                currentCategories={categories}
            />
            {categories.map((category) => (
                <GifGrid 
                    key={category} 
                    category={category}
                />
                    // <div key={category}>
                    //     <h3>{category}</h3>
                    //     <li>{category}</li>
                    // </div>
                )
            )}
        </>
    );
};
