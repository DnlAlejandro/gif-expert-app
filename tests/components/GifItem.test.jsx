import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("tests in <GifItem></GifItem>", () => {

    const title = "i'm title";
    const url = "https://image.com/";

    test("should be match with snapshot", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test("should show image with url and alt properly", () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
        //screen.debug();
    });

    test("should show title in component", () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

});
