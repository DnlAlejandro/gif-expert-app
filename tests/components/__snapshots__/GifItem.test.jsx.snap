// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <GifItem></GifItem> should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="i'm title"
      src="https://image.com/"
    />
    <p>
      i'm title
    </p>
  </div>
</div>
`;
